thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -594,7 +594,7 @@ void CDiskWriteSlaveActivityBase::process()
     StringBuffer tmpStr;
     fName.set(getPartFilename(*partDesc, 0, tmpStr).str());
     if (diskHelperBase->getFlags() & TDXtemporary && !container.queryJob().queryUseCheckpoints())
-        container.queryTempHandler()->registerFile(fName, container.queryOwner().queryGraphId(), usageCount, true);
+        tmpUsage.setown(container.queryTempHandler()->registerFile(fName, container.queryOwner().queryGraphId(), usageCount, true));
```

Comment:
agree, and missed this, and also initially did think it should just return an unlinked pointer, but erred on the side of caution of linking it here, forgeting about the other calls to registerFile which are not owning and would therefore leak.

I should be guaranteed to still be in scope when it gets to use this pointer in the same activity, the temp handler is owner at a higher scope.

So I would change it to return an unlinked pointer and store it in a plain member pointer instead of an Owned.